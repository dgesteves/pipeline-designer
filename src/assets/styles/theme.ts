import "styled-components";
import { DefaultTheme } from "styled-components";

const theme: DefaultTheme = {
  colors: {
    white: "#ffffff",
    background: "#1D1E22",
    violet: "#c43ad6",
    turquoise: "#30CFD0",
  },
  sizes: {
    size_1: "0.625rem",
    size_2: "0.75rem",
    size_3: "0.875rem",
    size_4: "1rem",
    size_5: "1.125rem",
    size_6: "1.25rem",
    size_7: "1.5rem",
    size_8: "1.875rem",
    size_9: "2rem",
  },
  gradients: {
    textGradient: "linear-gradient(to right, #30CFD0, #c43ad6)",
  },
  shadows: {
    textShadow:
      "0 0 80px rgb(192 219 255 / 35%), 0 0 32px rgb(65 120 255 / 24%)",
  },
};

export default theme;
