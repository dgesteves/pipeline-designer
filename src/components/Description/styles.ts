import styled from "styled-components";

export const Text = styled.article`
  text-shadow: ${({ theme }) => theme.shadows.textShadow};
  font-weight: 700;
  font-size: ${({ theme }) => theme.sizes.size_6};
  margin: ${({ theme }) => theme.sizes.size_5} 0
    ${({ theme }) => `calc(${theme.sizes.size_9}*2)`} 0;
  width: 60%;
  text-align: center;
  letter-spacing: 0.2rem;

  span {
    font-family: "Fredoka", sans-serif;
    background: ${({ theme }) => theme.gradients.textGradient};
    -webkit-background-clip: text;
    -webkit-text-fill-color: transparent;
    text-transform: capitalize;
  }
`;
