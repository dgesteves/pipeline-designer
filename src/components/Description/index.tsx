import { PIPELINE_DESIGNER } from "../../constants";
import { Text } from "./styles";

export default function Description() {
  return (
    <Text
      tabIndex={0}
      role="contentinfo"
      aria-label="Lorem ipsum text as placeholder"
    >
      Lorem ipsum dolor sit amet, <span>{PIPELINE_DESIGNER}</span> consectetur
      adipisicing elit. Amet consequuntur cum debitis dignissimos ducimus minima
      obcaecati praesentium, reiciendis <span>{PIPELINE_DESIGNER}</span> saepe
      veritatis voluptas voluptatibus? Blanditiis ipsam iste magnam neque nulla
      totam!
    </Text>
  );
}
