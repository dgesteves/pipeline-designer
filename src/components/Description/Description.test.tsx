import { render, screen } from "@testing-library/react";
import theme from "../../assets/styles/theme";
import { ThemeProvider } from "styled-components";
import GlobalStyles from "../../assets/styles/globalStyles";
import Description from "./index";

describe("Description component", () => {
  it("should render without crashing", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Description />
      </ThemeProvider>
    );

    expect(screen.getByRole("contentinfo")).toBeInTheDocument();
  });

  it("should have the correct text content", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Description />
      </ThemeProvider>
    );

    expect(screen.getByRole("contentinfo").textContent).toBe(
      "Lorem ipsum dolor sit amet, pipeline designer consectetur adipisicing elit. Amet consequuntur cum debitis dignissimos ducimus minima obcaecati praesentium, reiciendis pipeline designer saepe veritatis voluptas voluptatibus? Blanditiis ipsam iste magnam neque nulla totam!"
    );
  });

  it("should match previous snapshot", () => {
    render(
      <ThemeProvider theme={theme}>
        <GlobalStyles />
        <Description />
      </ThemeProvider>
    );

    expect(screen.getByRole("contentinfo")).toMatchSnapshot();
  });
});
