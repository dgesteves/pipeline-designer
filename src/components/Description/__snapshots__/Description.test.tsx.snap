// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Description component should match previous snapshot 1`] = `
<article
  aria-label="Lorem ipsum text as placeholder"
  class="sc-bczRLJ jMhavE"
  role="contentinfo"
  tabindex="0"
>
  Lorem ipsum dolor sit amet, 
  <span>
    pipeline designer
  </span>
   consectetur adipisicing elit. Amet consequuntur cum debitis dignissimos ducimus minima obcaecati praesentium, reiciendis 
  <span>
    pipeline designer
  </span>
   saepe veritatis voluptas voluptatibus? Blanditiis ipsam iste magnam neque nulla totam!
</article>
`;
